src/store/podcasts.ts

```diff
@@ -38,8 +40,26 @@ export const usePodcastStore = defineStore("podcasts", {
         image: "/default-podcast-image.jpg",
       },
     ],
+    readAccessPodcasts: [],
   }),
   getters: {
     getPodcasts: (state: State) => state.podcasts,
+    getReadAccessPodcasts: (state: State) => state.readAccessPodcasts,
+  },
+  actions: {
+    getReadAccessPodcast() {
+      this.readAccessPodcasts = []
+      const db = getFirestore()
+      this.podcasts.filter(async (podcast) => {
```

Comment:
no, thats not how an array filter works. you just return true/false for the things to keep in the array you don't need to push to another.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter